lib/ui/router/router_app.dart

```diff
@@ -14,56 +14,85 @@ import 'package:go_router/go_router.dart';
 
 part 'router_app.g.dart';
 
-@TypedShellRoute<RootRoute>(
-  routes: [
-    TypedGoRoute<HomeRoute>(
-      path: HomeRoute.path,
-    ),
-    TypedGoRoute<SessionsRoute>(
-      path: SessionsRoute.path,
+@TypedStatefulShellRoute<RootRoute>(
+  branches: [
+    // BottomNavigationBar Tabs
+    TypedStatefulShellBranch<HomeBranch>(
+      routes: [
+        TypedGoRoute<HomeRoute>(
+          path: HomeRoute.path,
+        ),
+      ],
     ),
-    TypedGoRoute<SponsorsRoute>(
-      path: SponsorsRoute.path,
+    TypedStatefulShellBranch<SessionsBranch>(
+      routes: [
+        TypedGoRoute<SessionsRoute>(
+          path: SessionsRoute.path,
+        ),
+      ],
     ),
-    TypedGoRoute<VenueRoute>(
-      path: VenueRoute.path,
+    TypedStatefulShellBranch<VenueBranch>(
+      routes: [
+        TypedGoRoute<VenueRoute>(path: VenueRoute.path),
+      ],
     ),
-    TypedGoRoute<ContributorsRoute>(
-      path: ContributorsRoute.path,
+    // Only Appears in Drawer and NavigationRail
+    TypedStatefulShellBranch<SponsorsBranch>(
+      routes: [
+        TypedGoRoute<SponsorsRoute>(
+          path: SponsorsRoute.path,
+        ),
+      ],
     ),
-    TypedGoRoute<SettingsRoute>(
-      path: SettingsRoute.path,
+    TypedStatefulShellBranch<ContributorsBranch>(
+      routes: [
+        TypedGoRoute<ContributorsRoute>(
+          path: ContributorsRoute.path,
+        ),
+      ],
     ),
-    TypedGoRoute<LicenseRoute>(
-      path: LicenseRoute.path,
+    TypedStatefulShellBranch<SettingsBranch>(
       routes: [
-        TypedGoRoute<AboutUsRoute>(
-          path: 'about-us',
+        TypedGoRoute<SettingsRoute>(
+          path: SettingsRoute.path,
         ),
-        TypedGoRoute<LegalNoticesRoute>(
-          path: 'legal-notices',
+      ],
+    ),
+    TypedStatefulShellBranch(
+      routes: [
+        TypedGoRoute<LicenseRoute>(
+          path: LicenseRoute.path,
+          routes: [
+            TypedGoRoute<AboutUsRoute>(
+              path: 'about-us',
+            ),
+            TypedGoRoute<LegalNoticesRoute>(
+              path: 'legal-notices',
+            ),
+          ],
         ),
       ],
     ),
   ],
 )
-class RootRoute extends ShellRouteData {
+class RootRoute extends StatefulShellRouteData {
```

Comment:
It looks like `$navigatorContainerBuilder` needs to be implemented.

https://pub.dev/documentation/go_router/latest/go_router/StatefulShellRoute-class.html

> When creating a StatefulShellRoute, a [navigatorContainerBuilder](https://pub.dev/documentation/go_router/latest/go_router/StatefulShellRoute/navigatorContainerBuilder.html) function must be provided. This function is responsible for building the actual container for the Widgets representing the branch Navigators. Typically, the Widget returned by this function handles the layout (including [Offstage](https://api.flutter.dev/flutter/widgets/Offstage-class.html) handling etc) of the branch Navigators and any animations needed when switching active branch.

https://github.com/flutter/packages/pull/4238/files#diff-b327c1cb82de061885a616b463f3dee7cb745e06fdb823b13185a4b56a5a1934R68

Reply:
Thank you, implemented. I don't use any animation for transition.
I think it's good welcome contribute issue to implement animation in transition.
[feat: impl navigatorContainerBuilder and use it](https://github.com/FlutterKaigi/conference-app-2023/pull/142/commits/a005121c7f9811a3e0f6ce301282f887a75dbfab)